packages/abi-coder/src/utils/getFunctionInputs.ts

```diff
@@ -0,0 +1,35 @@
+import { clone } from 'ramda';
+
+import type { JsonAbi, JsonAbiArgument } from '../types/JsonAbi';
+
+import { OPTION_REGEX, VOID_TYPE } from './constants';
+import { findTypeById } from './json-abi';
+
+export type ArgumentWithMetadata<TArg extends JsonAbiArgument = JsonAbiArgument> = TArg & {
+  isOptional: boolean;
+};
+
+export const getFunctionInputs = (params: {
+  jsonAbi: JsonAbi;
+  inputs: readonly JsonAbiArgument[];
+}): Array<ArgumentWithMetadata> => {
+  let inMandatoryRegion = false;
+  const inputs = clone(params.inputs);
```

Comment:
you can use `structuredClone` instead of `clone` here and remove the [dependency](https://github.com/FuelLabs/fuels-ts/pull/2777/files#diff-ae8b97dabdf446b7f229ac907def90a770ef59693a48fee298fdd926ab353d3dR36). I'm fine with this remaining `clone`, however, because I believe we are using `clone` in places where `structuredClone` can't be used because it doesn't clone functions, and so removing it or staying with it doesn't affect the `fuels` package.